File: docs-examples/example-java/src/test/java/io/micronaut/jms/docs/successhandler/SuccessHandlingSpec.java
package io.micronaut.jms.docs.successhandler;

import io.micronaut.jms.docs.AbstractJmsSpec;
import org.junit.jupiter.api.Test;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

public class SuccessHandlingSpec extends AbstractJmsSpec {

    @Test
    void testCustomSuccessHandlersAtTheMethodAndClassLevel() {
        SuccessHandlingProducer producer = applicationContext.getBean(SuccessHandlingProducer.class);
        SuccessHandlingConsumer consumer = applicationContext.getBean(SuccessHandlingConsumer.class);
        CountingSuccessHandler successHandler = applicationContext.getBean(CountingSuccessHandler.class);
        AccumulatingSuccessHandler classLevelSuccessHandler = applicationContext.getBean(AccumulatingSuccessHandler.class);

        producer.send("success message no. 1");
        producer.send("success message no. 2");

        await().atMost(5, SECONDS).until(() -> consumer.messages.size() == 2 &&
                successHandler.count.get() == 2 &&
                classLevelSuccessHandler.messages.size() == 2);
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/jms/docs/successhandler/SuccessHandlingSpec.kt
package io.micronaut.jms.docs.successhandler

import io.micronaut.jms.docs.AbstractJmsSpec
import org.junit.jupiter.api.Test

import org.awaitility.Awaitility.await
import java.util.concurrent.TimeUnit

class SuccessHandlingSpec : AbstractJmsSpec() {

    @Test
    fun testCustomSuccessHandlersAtTheMethodAndClassLevel() {
        val producer = applicationContext.getBean(SuccessHandlingProducer::class.java)
        val consumer = applicationContext.getBean(SuccessHandlingConsumer::class.java)
        val successHandler = applicationContext.getBean(CountingSuccessHandler::class.java)
        val classLevelSuccessHandler = applicationContext.getBean(AccumulatingSuccessHandler::class.java)

        producer.push("success message no. 1")
        producer.push("success message no. 2")

        await().atMost(5, TimeUnit.SECONDS).until {
            consumer.processed.size == 2 &&
                successHandler.count.get() == 2 &&
                classLevelSuccessHandler.messages.size == 2
        }
    }
}
